pkg/backend/cloud/client/client.go

```diff
@@ -506,7 +506,7 @@ func (pc *Client) PatchUpdateCheckpoint(ctx context.Context, update UpdateIdenti
 	}
 
 	req := apitype.PatchUpdateCheckpointRequest{
-		Version:    1,
+		Version:    2,
```

Comment:
I'll admit, it wasn't evident from reading this change why we need to rev the formats. It would seem all changes to the deployment/checkpoint structures are purely additive.

Reply:
Purely additive changes still require revving the checkpoint format if the interpretation of a resource by an old engine would be incorrect.  In this case, an old engine would ignore "external" and would then make incorrect decisions about how to interpret it.  Had we stored these "external" resources in a separate data structure in the checkpoint file, we may have been able to do this in a way which didn't have this problem, but with the approach we did take, it was necessary to rev the format.

Reply:
To restate: a change that is purely additive in a structural sense requires a version bump if its semantics are not backwards-compatible. First-class providers are another instance of this, which is why we folded the schema changes to that feature into final version of `ResourceV2` et. al.